dll/win32/advpack/files.c

```diff
@@ -341,17 +341,19 @@ static HRESULT DELNODE_recurse_dirtree(LPWSTR fname, DWORD flags)
         HANDLE hFindFile;
         WIN32_FIND_DATAW w32fd;
         BOOL done = TRUE;
+#ifdef __REACTOS__
         int fname_len;
 
-#ifdef __REACTOS__
         if (flags & ADN_DEL_IF_EMPTY)
-        {
             goto deleteinitialdirectory;
-        }
+#else
+        int fname_len = lstrlenW(fname);
 #endif
 
         /* Generate a path with wildcard suitable for iterating */
+#ifdef __REACTOS__
         fname_len = lstrlenW(fname);
+#endif
```

Comment:
I think this is less complicated and with fewer ReactOS guards (Lines 344-356):
```
        int fname_len = lstrlenW(fname);
#ifdef __REACTOS__
        if (flags & ADN_DEL_IF_EMPTY)
            goto deleteinitialdirectory;
#endif

        /* Generate a path with wildcard suitable for iterating */
```

Reply:
@whindsaks, Do you agree with this revert suggestion and explanation? As in, was your `lstrlenW()` change an optimization only?

Reply:
Reverted.